header-footer-grid/Traits/Core.php

```diff
@@ -90,6 +90,24 @@ public function sanitize_responsive_int_json( $input ) {
 			if ( isset( $filtered[ $key ] ) && is_numeric( $value ) ) {
 				$filtered[ $key ] = (int) $value;
 			}
+
+			// Optional if a suffix is present. We sanitize the array of suffixes below.
+			if ( 'suffix' === $key && is_array( $value ) ) {
+				$default_suffix    = array(
+					'mobile'  => 'px',
+					'tablet'  => 'px',
+					'desktop' => 'px',
+				);
+				$sanitized_devices = array();
+				foreach ( $value as $device => $suffix ) {
+					$filtered['suffix'][ $device ] = in_array( $suffix, array( 'px', 'em', 'rem', '%' ), true ) ? $suffix : 'px';
+					$sanitized_devices[]           = $device;
+				}
+
+				if ( array_diff( array_keys( $default_suffix ), $sanitized_devices ) ) {
+					$filtered['suffix'] = array_merge( $default_suffix, $filtered['suffix'] );
+				}
```

Comment:
I think we can simplify here. Apart from that, the current structure doesn't check suffix device, we can have it too.

```suggestion
				$filtered['suffix']  = array(
					'mobile'  => 'px',
					'tablet'  => 'px',
					'desktop' => 'px',
				);
	
				foreach ( $value as $device => $suffix ) {
					if( ! in_array( $suffix, array( 'px', 'em', 'rem', '%' ), true ) ) {
						continue;
					}
					
					// Additionally; device check was added. (for key check I think we can leverage expected keys from $filtered since we expect 1-1 match for devices between suffix and input devices)
					if( ! array_key_exists( $device, $filtered ) {
						continue;
					 } 
				
					$filtered['suffix'][ $device ] = $suffix;
				}
```

Reply:
I agree, your suggestion looks good. Thank you!